BP-TPSE-Projektgruppe-80/NaPyTau
# Add mypy, raise minimum version, fixup code

### Summary

This adds the mypy package as a dev-dependency. This allows us to statically analyze the code, to fully utilize this the analyzer is also added as a ci-step and any errors already in the code base are fixed

---

### PR checklist

- [x] I have written tests that fail without my changes. _Or I did not add them on purpose._
- [x] I have included instructions on how to test this, assuming they are not already apparent from the linked issue.

## How to test

As there is a new CI step added the static type checker tests itself, however if you want to verify its usage you can introduce some typing error in the project locally and then run the `typecheck` command
